ui/src/reusable_ui/components/dropdowns/DropdownButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {Component} from 'react'
+import classnames from 'classnames'
+import {ComponentColor, ComponentSize, IconFont} from 'src/reusable_ui/types'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import './DropdownButton.scss'
+
+interface Props {
```

Comment:
why are so many of these optional?

Reply:
Makes it easier to work with if you don't have to specify every aspect of the appearance. The defaults represent the most common appearance so you only have to specify when you want to use a non default appearance